src/Lykke.Service.BlockchainCashoutPreconditionsCheck/Controllers/BlackListController.cs

```diff
@@ -0,0 +1,181 @@
+        /// is address black listed
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet("{blockchainType}/{blockedAddress}")]
+        [ArgumentValidationExceptionFilter]
+        [SwaggerOperation("Get")]
+        [ProducesResponseType(typeof(BlackListResponse), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(void), (int)HttpStatusCode.NoContent)]
+        [ProducesResponseType(typeof(ErrorResponse), (int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType(typeof(ErrorResponse), (int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> GetBlackListAsync([FromRoute][Required] string blockchainType, [FromRoute][Required] string blockedAddress)
+        {
+            var model = await _blackListService.TryGetAsync(blockchainType, Trim(blockedAddress));
+
+            if (model == null)
+                return NoContent();
+
+            return Ok(Map(model));
+        }
+
+        /// <summary>
+        /// Take blocked addresses for specific blockchainType
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet("{blockchainType}")]
+        [ArgumentValidationExceptionFilter]
+        [SwaggerOperation("GetAll")]
+        [ProducesResponseType(typeof(BlackListEnumerationResponse), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(ErrorResponse), (int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType(typeof(ErrorResponse), (int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> GetAllAsync([FromRoute][Required] string blockchainType, [FromQuery] int take, [FromQuery] string continuationToken)
+        {
+            if (take <= 0)
+                throw new ArgumentValidationException("Field must be equal or greater than 0", $"{nameof(take)}");
+
+            var (models, newToken) = await _blackListService.TryGetAllAsync(blockchainType, take, continuationToken);
+
+            return Ok(new BlackListEnumerationResponse()
+            {
+                ContinuationToken = newToken,
+                List = models.Select(x => Map(x))
+            });
+        }
+
+        /// <summary>
+        /// Add black listed address
+        /// </summary>
+        /// <returns></returns>
+        [HttpPost]
+        [ArgumentValidationExceptionFilter]
+        [SwaggerOperation("Add")]
+        [ProducesResponseType(typeof(void), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(ErrorResponse), (int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType(typeof(ErrorResponse), (int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> AddBlockedAddressAsync([FromBody] AddBlackListModel request)
+        {
+            string blockedAddress = Trim(request.BlockedAddress);
+
+            BlackListModel model = new BlackListModel()
+            {
+                BlockedAddress = blockedAddress,
+                BlockedAddressLowCase = blockedAddress.ToLower(),
```

Comment:
ToLower shouldn't be here, even if BlockedAddressLowCase will be preserved. Move this to iether domain model - pass parameters via ctor or static factory method and initialize BlockedAddressLowCase there, on in the service - pass parameters to the SaveAsync and make new BlackListModel there.

Reply:
fixed